src/sql/workbench/services/objectExplorer/browser/objectExplorerViewTreeShim.ts

```diff
@@ -145,6 +146,28 @@ export class OEShimService extends Disposable implements IOEShimService {
 		}
 	}
 
+	/**
+	 * Refreshes a node - returning the refreshed set of children for that node.
+	 * @param node The node to refresh
+	 * @param viewId The ID of the view requesting the refresh
+	 */
+	public async refreshNode(node: ITreeItem, viewId: string): Promise<ITreeItem[]> {
+		if (node.payload) {
+			let sessionId = await this.getOrCreateSession(viewId, node);
+			let requestHandle = this.nodeHandleMap.get(generateNodeMapKey(viewId, node)) || node.handle;
+			let treeNode = new TreeNode(undefined, undefined, undefined, requestHandle, undefined, undefined, undefined, undefined, undefined, undefined);
+			treeNode.connection = new ConnectionProfile(this.capabilities, node.payload);
+			return this.oe.refreshTreeNode({
+				success: undefined,
+				sessionId,
+				rootNode: undefined,
+				errorMessage: undefined
+			}, treeNode).then(e => e.map(n => this.treeNodeToITreeItem(viewId, n, node)));
```

Comment:
Any reason not to do await here?

Reply:
This was just copy-pasted with a minor change so I didn't look over it, I'll refactor to be cleaner though.